couchbase/tests/test_unit.py

```diff
@@ -91,3 +91,35 @@ def test_config(test_case, dd_run_check, extra_config, expected_http_kwargs, ins
         )
         http_wargs.update(expected_http_kwargs)
         r.get.assert_called_with('http://localhost:8091/pools/default/tasks', **http_wargs)
+
+
+def test_extract_index_tags(instance):
+    couchbase = Couchbase('couchbase', {}, [instance])
+    """
+    Test to ensure that tags are extracted properly from keyspaces. Takes into account the different
+    forms of the keyspace and extract the tags from them accordingly. Docs:
+    https://docs.couchbase.com/server/current/rest-api/rest-index-stats.html#responses-3
+    https://docs.couchbase.com/server/current/n1ql/n1ql-language-reference/createprimaryindex.html#keyspace-ref
+    """
+    EXTRACTED_INDEX_TAGS_TEST_SAMPLES = {
```

Comment:
nit- to make this test a bit simpler, you can use `pytest.mark.parametrize`. An example is this test:
https://github.com/DataDog/integrations-core/blob/master/couchbase/tests/test_unit.py#L62